src/main/java/in/projecteka/consentmanager/clients/ClientRegistryClient.java

```diff
@@ -62,15 +62,42 @@ public ClientRegistryClient(WebClient.Builder webClientBuilder, String baseUrl)
                 .bodyToMono(Session.class);
     }
 
+    public Mono<Session> getTokenWithRefreshToken(String clientId, String clientSecret, String refreshToken) {
+        return webClientBuilder.build()
+                .post()
+                .uri("/api/1.0/sessionsRefresh")
+                .contentType(APPLICATION_JSON)
+                .accept(APPLICATION_JSON)
+                .body(BodyInserters.fromValue(requestWithRefreshToken(clientId, clientSecret, refreshToken)))
+                .retrieve()
+                .onStatus(HttpStatus::isError, clientResponse -> clientResponse.bodyToMono(Properties.class)
+                        .doOnNext(properties -> logger.error(properties.toString()))
+                        .thenReturn(ClientError.unAuthorized()))
+                .bodyToMono(Session.class);
+    }
+
     private SessionRequest requestWith(String clientId, String clientSecret) {
         return new SessionRequest(clientId, clientSecret, "password");
     }
 
+    private SessionRequestWithRefreshToken requestWithRefreshToken(String clientId, String clientSecret, String refreshToken) {
+        return new SessionRequestWithRefreshToken(clientId, clientSecret, "refresh_token", refreshToken);
+    }
+
     @AllArgsConstructor
     @Data
     private static class SessionRequest {
         private String clientId;
         private String clientSecret;
         private String grantType;
     }
+
+    @AllArgsConstructor
+    @Data
```

Comment:
Can't we use \`@Value\` here instead of having mutable classes? applies for the all the classes you have created as part of this PR. There are quite a few APIs I have modified to use \`@Value\` , you can check that.

Reply:
done